### DISCUSSION TITLE
Bar Chart or Pie Chart

### DISCUSSION THREAD

**Discussion by jyasuna** [GSI-CS-CO/chart-fx#422] [General]
I am migrating from JFreechrat to Chartfx. 
Line and Scatter plots all set

Wondering if there are renderers for Bar Charts and Pie Charts?
(didn't see them, so thought to ask)

**Comment by jyasuna** [comment]
looking for something like these 2 ...
![bar](https://user-images.githubusercontent.com/3989992/119861645-9e722100-bee5-11eb-9223-127dc83b7509.png)
![pie](https://user-images.githubusercontent.com/3989992/119861664-a336d500-bee5-11eb-9dc3-2b56fdeadee6.png)

**Reply by RalphSteinhagen** [reply]
Hey, thanks for your interest. Chart-fx implements bar-type plots in various forms, with and without error bars, all bars coloured the same or with some that are highlighted, ...

![image](https://user-images.githubusercontent.com/46007894/119869588-b31fda80-bf20-11eb-8ea3-c3d18b2a74fb.png)
[HistogramRendererTests.java](https://github.com/GSI-CS-CO/chart-fx/blob/master/chartfx-chart/src/test/java/de/gsi/chart/renderer/spi/HistogramRendererTests.java)

or
![image](https://user-images.githubusercontent.com/46007894/119869742-e2364c00-bf20-11eb-992f-848118b587e1.png)
[ErrorDataSetRendererStylingSample.java](https://github.com/GSI-CS-CO/chart-fx/blob/master/chartfx-samples/src/main/java/de/gsi/chart/samples/ErrorDataSetRendererStylingSample.java)

Where you can also test/experiment with the various plotting settings.

We haven't implemented a pie-chart though. Until now we mostly focused on real-time display of measurement data. Implementing a pie-chart (relying on the same `DataSet` interface) shouldn't be hard to implement though or -- alternatively -- you could also try the stock JavaFX pie chart -- for low number of data points and update rates that implementation should be perfectly fine. 

PRs and extensions are of course welcome.

Hope this helps.
